.github/workflows/check.yml

```diff
@@ -13,34 +13,98 @@ jobs:
         image: mysql:5.7
         options: --health-cmd "mysqladmin ping -h localhost" --health-interval 20s --health-timeout 10s --health-retries 10
         ports:
-          - "3306:3306"
+        - "3306:3306"
         env:
           MYSQL_ROOT_PASSWORD: root
           MYSQL_USER: ci
           MYSQL_PASSWORD: password
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up JDK 8
-        uses: actions/setup-java@v3
-        with:
-          java-version: 8
-          distribution: 'temurin'
-          cache: "gradle"
-      - name: Connect
-        run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "show databases;"
-      - name: Create database
-        run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "create database ci_test;"
-      - name: Build with testing
-        run: ./gradlew --stacktrace :embulk-output-mysql:check
+    - uses: actions/checkout@v4
+    - name: Set up JDK 8
+      uses: actions/setup-java@v4
+      with:
+        java-version: 8
+        distribution: 'zulu'
+        cache: "gradle"
+    - name: Connect
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "show databases;"
+    - name: Create database
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "create database ci_test;"
+    - name: Build with testing
+      run: ./gradlew --stacktrace :embulk-output-mysql:check
+      env:
+        _JAVA_OPTIONS: "-Xmx2048m -Xms512m"
+        EMBULK_OUTPUT_MYSQL_TEST_CONFIG: "${{ github.workspace }}/ci/mysql.yml"
+    - uses: actions/upload-artifact@v4
+      if: always()
+      with:
+        name: mysql5_7
+        path: embulk-output-mysql/build/reports/tests/test
+  mysql8_3:
+    runs-on: ubuntu-latest
+    # push: always run.
+    # pull_request: run only when the PR is submitted from a forked repository, not within this repository.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    strategy:
+      fail-fast: false
+    services:
+      mysql:
+        # Due to MySQL 8.4 disabled mysql_native_password by default,
+        # Connector/J 5.x can't connect to database.
+        # So, Use MySQL 8.3.
```

Comment:
```suggestion
        # Testing with MySQL 8.3 here, neither with 8.4 nor 9.0, because
        # MySQL 8.4 has disabled mysql_native_password by default.
        # Connector/J 5.x cannot connect to MySQL 8.4 due to this.
        # TODO: Start testing with MySQL 8.4 and/or 9.0 with Connector/J 8.x.
```